### PR TITLE
Add user OpenApi [SFEQS-726]

### PR DESCRIPTION
**Pull request by grausof**
This PR begins to describe the openapi specification for the user. It introduces a new `Signature` entity that keeps track of all the signatures affixed to the various fields and integration with the QTSP

### CODE REVIEW

**Comment by gunzip** [comment]
lgtm (I don't have enough context to enter into details)

File: docs/openapi-user.yaml [hunk 1]
```diff
@@ -1,170 +1,419 @@
 openapi: 3.0.3
-
 info:
-  title: Firma con IO User APIs
-  description: Any APIs called by IO App
-  version: 1.0.0
-
+  title: IO Sign - User API
```
Comment (hevelius) [inline docs/openapi-user.yaml:3]: What do you think about using "Firma con IO - User API"?

File: docs/openapi-user.yaml [hunk 2]
```diff
@@ -1,170 +1,419 @@
...
+      format: base64url
+      description: SPID SAML assertion encoded in base64
+
+    ClausesSignature:
+      type: object
+      properties:
+        signatureFieldId:
+          type: string
+        accepted:
+          type: boolean
```
Comment (hevelius) [inline docs/openapi-user.yaml:320]: The clauses must not also have a required field boolean? Should the document clauses in the UI not be mandatory or optional?

File: docs/openapi-user.yaml (hunk 1 above)
Comment (grausof) [reply docs/openapi-user.yaml:3]: I agree. Thank you!

Comment (grausof) [reply docs/openapi-user.yaml:3]: Other references also have the name "io-sign" including the repository. For now I would leave it like this!

File: docs/openapi-user.yaml (hunk 2 above)
Comment (grausof) [reply docs/openapi-user.yaml:320]: Yes, to show them you have to retrieve them from the `signature-request`. In this case it is the POST of the accepted clauses where it is not necessary to know if they are mandatory or not.

Comment (hevelius) [reply docs/openapi-user.yaml:320]: Great! Thanks!

File: docs/openapi-user.yaml (hunk 1 above)
Comment (hevelius) [reply docs/openapi-user.yaml:3]: In the context of IO the name "Firma con IO" is used I thought it was better to refer to this :-D

**Review by hevelius — APPROVED** [review]
LGTM

**Review by lucacavallaro — APPROVED** [review]

**Pull request merged**